genshin/client.py

```diff
@@ -120,16 +121,15 @@ def session(self) -> aiohttp.ClientSession:
 
         return self._session
 
-    @property
-    def cookies(self) -> Mapping[str, str]:
-        """The cookie jar belonging to the current session"""
-        return {cookie.key: cookie.value for cookie in self.session.cookie_jar}
+    @session.setter
+    def session(self, session: aiohttp.ClientSession) -> None:
+        if not isinstance(session, aiohttp.ClientSession):
+            raise TypeError(f"session must be of type aiohttp.ClientSession, got {type(session)}")
+
+        if self._session is not None:
+            self._session.close()
```

Comment:
Realized that this won't work as `close` is a coro, should have actually tested this. Should I instead raise a RuntimeError or something similar, or just silently overwrite it and have aiohttp get angry?

Reply:
Start an asyncio task